src/queue_pair.rs

```diff
@@ -312,6 +312,12 @@ impl QueuePair {
         let mut sr = SendWr::new_send(lms, wr_id, imm);
```

Comment:
The struct `SendWr` is dropped at the end of fn `submit_send`. There is a vec of sge in `SendWr` which is used in `ibv_post_send`. Does `ibv_post_send` copy the vec? If it doesn't, then there is a potential UAF bug.

<https://man7.org/linux/man-pages/man3/ibv_post_send.3.html#NOTES>
>   The **buffers** used by a WR can only be safely reused after WR the
       request is fully executed and a work completion has been
       retrieved from the corresponding completion queue (CQ). However,
       if the IBV_SEND_INLINE flag was set, the buffer can be reused
       immediately after the call returns.

I'm not sure whether the "buffers" include the s/g array.

Reply:
I had the same doubts as you before. 
So I checked the source code of rdma-core, and found that `ibv_post_send` will `memcpy` sg_list.